### PR TITLE
Amprokop backfill ramp tasks

### PR DESCRIPTION
**Pull request by amprokop**
Resolves https://github.com/department-of-veterans-affairs/caseflow/issues/9337

### CODE REVIEW

File: app/models/tasks/judge_task.rb [hunk 1]
```diff
@@ -22,67 +22,53 @@ def previous_task
...
-    return false if task.appeal.hearing_docket?
-
-    # If it's an evidence submission case, we need to wait until the
-    # evidence submission window is over
-    if task.appeal.evidence_submission_docket?
-      return false if task.appeal.receipt_date > 90.days.ago
-    end
 
-    task.children.all? { |t| !t.is_a?(JudgeTask) && !t.active? }
+    task.children.all? { |t| !t.is_a?(JudgeTask) }
```
Comment (amprokop) [inline app/models/tasks/judge_task.rb]: This is a change — for every Appeal without a JudgeTask, we need to backfill with DistributionTasks — even if that appeal has an open mail task or an open judge task.

File: app/models/tasks/judge_task.rb [hunk 2]
```diff
@@ -22,67 +22,53 @@ def previous_task
...
-      )
-      Rails.logger.info("Assigned judge task with task id #{task.id} to #{task.assigned_to.css_id}")
+  def self.backfill_tasks(root_tasks)
+    transaction do
+      root_tasks.each do |root_task|
+        Rails.logger.info("Creating subtasks for appeal #{root_task.appeal.id}")
+        RootTask.create_subtasks!(root_task.appeal, root_task)
+        distribution_task = DistributionTask.find_by(parent: root_task)
+        # Update any open IHP tasks if they exist so that they block distribution.
+        ihp_task = InformalHearingPresentationTask.find_by(appeal: root_task.appeal)
```
Comment (cmgiven) [inline app/models/tasks/judge_task.rb]: Do we need to do this with mail tasks too? Any other kinds of tasks we should move over?

Comment (amprokop) [reply app/models/tasks/judge_task.rb]: We don't need to do it with mail tasks — not implemented yet, but the approach we agreed on was for me to modify the `ready_for_distribution_at` to exclude mail tasks.

Comment (amprokop) [reply app/models/tasks/judge_task.rb]: @lowellrex — would you mind taking a quick look at this — as far as I know, IHP and mail tasks are the only currently used task types that should block distribution — are there any others?

**Pull request merged**